stake_vote/src/lib.rs

```diff
@@ -36,7 +37,9 @@ pub async fn start_stakes_and_votes_loop(
     rpc_client: Arc<RpcClient>,
     grpc_url: String,
 ) -> anyhow::Result<tokio::task::JoinHandle<()>> {
+    log::info!("Start Stake and Vote loop.");
     let mut account_gyzer_stream = subscribe_geyzer(grpc_url).await?;
+    log::info!("Stake and Vote geyzer subscription done.");
     let jh = tokio::spawn(async move {
```

Comment:
Just return here, or preferably make the function async and let the caller decide whether to spawn a new task or not. I understand there is a loop here

Reply:
I understand the idea that we can decide how to start a task. I think here it's a whole. There's some init and then the loop task must be started. If not, there's no need to call the method. 
The other possibility is to use a struct that has an init method and a start loop method. It adds complexity and add the possibility to start the loop without doing the init. 
I think it's the same as RPC or Geyser subscription (https://github.com/blockworks-foundation/lite-rpc/blob/fa0433f74a3e0b89d5d93dd5fc4892e08dc0d1e9/cluster-endpoints/src/grpc_subscription.rs#L272). It's part the implementation and can't be removed.